operator/src/trustee.rs

```diff
@@ -127,61 +133,82 @@ pub async fn generate_kbs_configurations(
-            }
-        })
-        .collect::<Result<Vec<_>, _>>()?;
-    let reference_values_json = serde_json::to_string(&reference_values)?;
-
-    let mut data = BTreeMap::new();
-    data.insert(
-        "reference-values.json".to_string(),
-        reference_values_json.to_string(),
-    );
+    for (flag, path_suffix) in [
+        ("kernels", "usr/lib/modules"),
+        ("esp", "usr/lib/bootupd/updates"),
+    ] {
+        let full_path = mountpoint.join(path_suffix);
+        add_flag(flag, full_path.to_str().unwrap());
+    }
+    for (flag, value) in [
+        ("configmap", name),
+        ("namespace", trustee_namespace),
+        ("efivars", "/reference-values/efivars/qemu-ovmf/fcos-42"),
+        ("mokvars", "/reference-values/mok-variables/fcos-42"),
+    ] {
+        add_flag(flag, value);
+    }
 
-    let config_map = ConfigMap {
-        metadata: kube::api::ObjectMeta {
-            name: Some(name.to_string()),
-            namespace: Some(namespace.to_string()),
+    let pod_spec = PodSpec {
+        service_account_name: Some("compute-pcrs".to_string()),
+        containers: vec![Container {
+            name: job_name.to_string(),
+            image: Some(pcrs_compute_image.to_string()),
+            command: Some(cmd),
+            volume_mounts: Some(vec![VolumeMount {
+                name: volume_name.to_string(),
+                mount_path: mountpoint.to_str().unwrap().to_string(),
+                ..Default::default()
+            }]),
+            ..Default::default()
+        }],
+        volumes: Some(vec![Volume {
+            name: volume_name.to_string(),
+            image: Some(ImageVolumeSource {
+                reference: Some(BOOT_IMAGE.to_string()),
+                ..Default::default()
+            }),
+            ..Default::default()
+        }]),
+        restart_policy: Some("Never".to_string()),
+        ..Default::default()
+    };
+    let job = Job {
+        metadata: ObjectMeta {
+            name: Some(job_name.to_string()),
+            namespace: Some(job_namespace.to_string()),
             ..Default::default()
         },
-        data: Some(data),
+        spec: Some(JobSpec {
+            template: PodTemplateSpec {
+                spec: Some(pod_spec),
+                ..Default::default()
+            },
+            ..Default::default()
+        }),
         ..Default::default()
     };
 
-    let config_maps: Api<ConfigMap> = Api::namespaced(client, namespace);
-    let create = config_maps
-        .create(&PostParams::default(), &config_map)
-        .await;
-    info_if_exists!(create, "ConfigMap", name);
-
+    let jobs: Api<Job> = Api::namespaced(client.clone(), job_namespace);
+    let create = jobs.create(&PostParams::default(), &job).await;
+    info_if_exists!(create, "Job", job_name);
+    let completed = await_condition(jobs.clone(), job_name, is_job_completed());
+    let _ = timeout(Duration::from_secs(900), completed).await?;
```

Comment:
Is this blocking until it completes?
The usual way in kubernetes is watching objects with a controller and then trigger a reconciliation loop if there is a change in the state. Can we implement something like this here? See the rust documentation for the [Controller](https://docs.rs/kube/1.1.0/kube/runtime/struct.Controller.html)